src/ng/browser.js

```diff
@@ -235,10 +235,22 @@ function Browser(window, document, $log, $sniffer) {
 
   // This variable should be used *only* inside the cacheState function.
   var lastCachedState = null;
+  function getCurrentState() {
+    try {
+      var state = history.state;
+      if (state === void 0) {
```

Comment:
It's not going to be either a valid object, undefined or null, eg https://dxr.mozilla.org/mozilla-central/source/dom/webidl/History.webidl#18